driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -209,4 +209,119 @@ mod tests {
+            batch_id: 0,
+        }));
+        orderbook.handle_event_data(deposit_1, 1, 0, H256::zero(), 0);
+        let deposit_2 = EventData::Added(Event::Deposit(Deposit {
+            user: Address::from_low_u64_be(2),
+            token: Address::from_low_u64_be(0),
+            amount: 1.into(),
+            batch_id: 0,
+        }));
+        orderbook.handle_event_data(deposit_2, 2, 0, H256::zero(), 0);
+
+        let auction_data = orderbook.get_auction_data(1.into()).unwrap();
+        assert_eq!(
+            auction_data.0.read_balance(0, Address::from_low_u64_be(2)),
+            3
+        );
+        orderbook.delete_events_starting_at_block(1);
+        let auction_data = orderbook.get_auction_data(1.into()).unwrap();
+        assert_eq!(
+            auction_data.0.read_balance(0, Address::from_low_u64_be(2)),
+            1
+        );
+    }
+
+    #[test]
+    fn events_get_sorted() {
+        // We add a token, deposit that token, request to withdraw that token, withdraw it but give
+        // the events to the orderbook in a shuffled order.
+        let token_listing_0 = EventData::Added(Event::TokenListing(TokenListing {
+            token: Address::from_low_u64_be(0),
+            id: 0,
+        }));
+        let token_listing_1 = EventData::Added(Event::TokenListing(TokenListing {
+            token: Address::from_low_u64_be(1),
+            id: 1,
+        }));
+        let order_placement = EventData::Added(Event::OrderPlacement(OrderPlacement {
+            owner: Address::from_low_u64_be(2),
+            index: 0,
+            buy_token: 1,
+            sell_token: 0,
+            valid_from: 0,
+            valid_until: 10,
+            price_numerator: 10,
+            price_denominator: 10,
+        }));
+        let deposit = EventData::Added(Event::Deposit(Deposit {
+            user: Address::from_low_u64_be(2),
+            token: Address::from_low_u64_be(0),
+            amount: 10.into(),
+            batch_id: 0,
+        }));
+        let withdraw_request = EventData::Added(Event::WithdrawRequest(WithdrawRequest {
+            user: Address::from_low_u64_be(2),
+            token: Address::from_low_u64_be(0),
+            amount: 5.into(),
+            batch_id: 0,
+        }));
+        let withdraw = EventData::Added(Event::Withdraw(Withdraw {
+            user: Address::from_low_u64_be(2),
+            token: Address::from_low_u64_be(0),
+            amount: 3.into(),
+        }));
+
+        let mut orderbook = Orderbook::default();
+        orderbook.handle_event_data(token_listing_1, 0, 1, H256::zero(), 0);
+        orderbook.handle_event_data(order_placement, 0, 2, H256::zero(), 0);
+        orderbook.handle_event_data(withdraw, 2, 0, H256::zero(), 300);
+        orderbook.handle_event_data(deposit, 0, 3, H256::zero(), 0);
+        orderbook.handle_event_data(withdraw_request, 1, 0, H256::zero(), 0);
+        orderbook.handle_event_data(token_listing_0, 0, 0, H256::zero(), 0);
```

Comment:
In order to make this test more minimal, how about simply placing an order before listing it and asserting we can create state without an error? Would that be assuming too much about the internal consistency checks?

Reply:
I'm assuming you mean placing an order before listing the tokens. In that case like you said I thought it was assuming too much about the consistency check so I wanted to go with something measurable in the orderbook like a balance or an order used amount.